### ISSUE TITLE
using `--lua-version=5.1` on windows still uses `5.4`

### ISSUE THREAD

**Issue by Frityet** [luarocks/luarocks#1611]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions in the [Gitter
  chat-room](https://gitter.im/luarocks/luarocks).
- Please do NOT submit duplicate bug reports. Look through the [issue
  tracker](https://github.com/luarocks/luarocks/issues) before submitting.

### Please fill out the following:
- **Platform:**
  - Windows/Linux/MacOS.
 Windows

- **LuaRocks version:**
  - `luarocks --version`
Latest (dev) pulled 22/02/24

- **Configuration file:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find
    the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.
https://gist.github.com/Frityet/9dbc00cccf52c278130874d3de2c3be3

- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.
```yaml
fs.current_dir()
fs.current_dir()
fs.is_superuser()
fs.change_dir_to_root()
fs.current_dir()
fs.is_dir("C:\\Users\\Frit")
fs.absolute_name("C:\\Users\\Frit")
fs.current_dir()

Error: Lua 5.1 interpreter not found at C:\Program Files\Lua\5.4
Tried:	C:\Program Files\Lua\5.4\bin\lua5.1.exe (C:\Program Files\Lua\5.4\bin\lua5.1.exe: No such file or directory)
	C:\Program Files\Lua\5.4\bin\lua51.exe (C:\Program Files\Lua\5.4\bin\lua51.exe: No such file or directory)
	C:\Program Files\Lua\5.4\bin\lua-5.1.exe (C:\Program Files\Lua\5.4\bin\lua-5.1.exe: No such file or directory)
	C:\Program Files\Lua\5.4\bin\lua-51.exe (C:\Program Files\Lua\5.4\bin\lua-51.exe: No such file or directory)
	C:\Program Files\Lua\5.4\bin\luajit.exe (C:\Program Files\Lua\5.4\bin\luajit.exe: No such file or directory)
	C:\Program Files\Lua\5.4\lua5.1.exe (C:\Program Files\Lua\5.4\lua5.1.exe: No such file or directory)
	C:\Program Files\Lua\5.4\lua51.exe (C:\Program Files\Lua\5.4\lua51.exe: No such file or directory)
	C:\Program Files\Lua\5.4\lua-5.1.exe (C:\Program Files\Lua\5.4\lua-5.1.exe: No such file or directory)
	C:\Program Files\Lua\5.4\lua-51.exe (C:\Program Files\Lua\5.4\lua-51.exe: No such file or directory)
	C:\Program Files\Lua\5.4\luajit.exe (C:\Program Files\Lua\5.4\luajit.exe: No such file or directory)
	C:\Program Files\Lua\5.4\lua.exe (C:\Program Files\Lua\5.4\lua.exe: No such file or directory)
Please set your Lua interpreter with:

   luarocks --lua-version=5.1 --local config variables.LUA <d:\path\lua.exe>
```

- **Description:**
  - The steps you took in plain English to reproduce the problem.
I created a `config-5.1.lua` next to my `config-5.4.lua` so I could use luajit, but when I try an install a module it seems to keep using lua 5.4. Setting `LUA = "luajit"` in the config-5.1.lua does not work either

**Comment by Frityet** [comment]
> mmm, let's see...
> 
> ok, so your 5.1 config from the gist has no Lua 5.4 references whatsoever, which is good.
> 
>     * How did you get that config output, was it just running `luarocks config --lua-version=5.1`?
> 
> 
> ```
> Error: Lua 5.1 interpreter not found at C:\Program Files\Lua\5.4
> ```
> 
>     * which command did you run to get that error above?
> 
>     * what is the full path and filename of your Lua 5.1 interpreter?
> 
>     * is your Lua 5.1 interpreter available in the `PATH`?

I did `luarocks --lua-version=5.1 config`, and yes it is all in path, as `luajit -v` works well,

>what is the full path and filename of your Lua 5.1 interpreter?
`C:\Program Files\Lua\5.1\bin\luajit.exe`

**Comment by Frityet** [comment]
![image](https://github.com/luarocks/luarocks/assets/25426233/7664b280-3e1c-4095-b3fc-afb3e3bea63a)

**Comment by hishamhm** [comment]
I think I found the issue. I believe it is picking that `c:\program files\lua\5.4` default path from `luarocks/core/hardcoded.lua` which is generated by `install.bat`, and then that is preventing it from looking up through the `PATH` for looking for a Lua 5.1 interpreter. I'm opening a PR now!

**Comment by Frityet** [comment]
> I think I found the issue. I believe it is picking that `c:\program files\lua\5.4` default path from `luarocks/core/hardcoded.lua` which is generated by `install.bat`, and then that is preventing it from looking up through the `PATH` for looking for a Lua 5.1 interpreter. I'm opening a PR now!

Perfect! ill try that out, thank you!
